packages/theme/src/cli/utilities/theme-environment/local-assets.ts

```diff
@@ -94,35 +94,143 @@ function isCompiledAssetRequest(event: H3Event) {
 function handleCompiledAssetRequest(event: H3Event, ctx: DevServerContext) {
   const assetPath = parseServerEvent(event).pathname.split('/').at(-1)
 
-  if (assetPath === 'styles.css') {
-    const allLiquidFiles = [...ctx.localThemeFileSystem.files.entries()].filter(([key]) => key.endsWith('.liquid'))
+  switch (assetPath) {
+    case 'styles.css':
+      return handleStylesCss(ctx, event)
+    case 'block-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'block')
+    case 'snippet-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'snippet')
+    case 'section-scripts.js':
+      return handleBlockScriptsJs(ctx, event, 'section')
+    default:
+      return handleError(event)
```

Comment:
Maybe we should let it go to the server in this case instead of throwing? In case we introduce a new type or something like that?